pgrx/src/spi.rs

```diff
@@ -324,9 +324,9 @@ fn prepare_datum(datum: Option<pg_sys::Datum>) -> (pg_sys::Datum, std::os::raw::
     }
 }
 
-impl<'a> Query for &'a str {
+impl<'conn> Query<'conn> for &str {
```

Comment:
Hmm. How long *does* the `&str` need to live (or not-live), exactly?

Reply:
The `&str` (ie, `Self`) only needs to live as long as it takes for us to copy it into a `CString` to send off to Postgres as the query string.  Same us true for the `impl Query for &String` impl.  

I wonder if we should have instead just done:  `impl<'conn, Q: AsRef<str>> Query<'conn> for Q` to better express what we're implementing for.

tl;dr is that we don't care how long the `&str` lives b/c we copy it.

Reply:
Ah okay.